Source/ASRunLoopQueue.mm

```diff
@@ -377,17 +372,13 @@ - (instancetype)init
     // __unsafe_unretained allows us to avoid flagging the memory cycle detector.
     __unsafe_unretained __typeof__(self) weakSelf = self;
     void (^handlerBlock) (CFRunLoopObserverRef observer, CFRunLoopActivity activity) = ^(CFRunLoopObserverRef observer, CFRunLoopActivity activity) {
+      while (_internalQueue.count > 0) {
```

Comment:
Doesn't this defeat the point of the `weakSelf`?

Reply:
Addressed